detox/ios/Detox/DetoxInstrumentsManager.m

```diff
@@ -252,21 +252,23 @@ - (id)_configForDetoxRecordingWithURL:(NSURL*)URL
 	[config setRecordThreadInformation:YES];
 	[config setCollectStackTraces:YES];
 	[config setSymbolicateStackTraces:YES];
-	[config setSamplingInterval:0.1];
+	[config setSamplingInterval:samplingInterval];
 	
 	return config;
 }
 
-- (void)startRecordingAtURL:(NSURL*)URL
+- (void)startRecordingAtURL:(NSURL*)URL andSamplingInterval:(NSTimeInterval)samplingInterval
 {
-	dtx_log_info(@"Starting recording at %@", URL);
-	[_recorderInstance startProfilingWithConfiguration:[self _configForDetoxRecordingWithURL:URL]];
+	dtx_log_info(@"Starting recording at %@ with samplingInterval %f", URL, samplingInterval);
+	id config = [self _configForDetoxRecordingWithURL:URL andSamplingInterval:samplingInterval];
```

Comment:
This should be moved either on the outside, if the method was changed to `startRecordingWithConfiguration :`, or this method should receive an `NSDictionary` with configuration and translate that while creating the `DTXProfilingConfiguration` object.

Reply:
Agreed